tests/snmp/test_snmp_phy_entity.py

```diff
@@ -786,11 +786,23 @@ def redis_get_keys(duthost, db_id, pattern):
     :param pattern: Redis key pattern
     :return: A list of key name in string
     """
+    totalOutput = []
+
+    def run_cmd_store_output(cmd):
+        logging.debug('Getting keys from redis by command: {}'.format(cmd))
+        output = duthost.shell(cmd)['stdout'].strip()
+        if output:
+            totalOutput.extend(output.split('\n'))
+
+    if duthost.is_multi_asic:
+        # Search the namespaces as well on LCs
+        for asic in duthost.frontend_asics:
+            cmd = 'sonic-db-cli -n {} {} KEYS \"{}\"'.format(asic.namespace, db_id, pattern)
+            run_cmd_store_output(cmd)
+
     cmd = 'sonic-db-cli {} KEYS \"{}\"'.format(db_id, pattern)
-    logging.debug('Getting keys from redis by command: {}'.format(cmd))
-    output = duthost.shell(cmd)
-    content = output['stdout'].strip()
-    return content.split('\n') if content else None
+    run_cmd_store_output(cmd)
```

Comment:
line 803 and 804 should be inside "else"  and need not be executed on the host for multi-asic?

Reply:
Some keys are stored in the global namespace, others are stored in the per-asic namespace on multi-asic systems.
E.G. CPU Temp info will be in the global namespace:
```
> sonic-db-cli STATE_DB keys \* | grep -i "TEMPERATURE_INFO"
TEMPERATURE_INFO|Cpu temp sensor
TEMPERATURE_INFO|Cpu SBTSI

> sonic-db-cli -n asic0 STATE_DB keys \* | grep -i "TEMPERATURE_INFO"
ASIC_TEMPERATURE_INFO
```